core/src/engine/single_waiter_mutex.cpp

```diff
@@ -0,0 +1,29 @@
+#include <userver/engine/single_waiter_mutex.hpp>
+
+#include <engine/impl/mutex_impl.hpp>
+
+USERVER_NAMESPACE_BEGIN
+
+namespace engine {
+
+struct SingleWaiterMutex::Impl final {
+  impl::MutexImpl<impl::WaitListLight> mutex;
```

Comment:
it's not a mutex, it is an impl or mutex_impl (impl_->mutex_impl is too cryptic too, though)

Reply:
Ok. I try to  rename  Impl -> ImplWrapper  
and change 
`impl_->mutex.lock()`
to 
`(**impl_wrapper_).lock()`